src/Components/Aspire.Microsoft.EntityFrameworkCore.SqlServer/AspireSqlServerEFCoreSqlClientExtensions.cs

```diff
@@ -113,6 +113,42 @@ void ConfigureRetry()
 #pragma warning disable EF1001 // Internal EF Core API usage.
             if (!settings.DisableRetry || settings.CommandTimeout.HasValue)
             {
+#if NET9_0_OR_GREATER
+                builder.CheckDbContextRegistered<TContext>();
+
+                builder.Services.ConfigureDbContext<TContext>(optionsBuilder => optionsBuilder.ConfigureSqlEngine(options =>
+                {
+                    var extension = optionsBuilder.Options.FindExtension<SqlServerOptionsExtension>();
```

Comment:
Why does this one have all this code duplicated between the 2 TFMs? The other ones are able to share code.